src/apm.js

```diff
@@ -26,55 +25,57 @@ module.exports = {
     return path.join(this.getAtomDirectory(), '.apm');
   },
 
-  getResourcePath(callback) {
-    if (process.env.ATOM_RESOURCE_PATH) {
-      return process.nextTick(() => callback(process.env.ATOM_RESOURCE_PATH));
-    }
-
-    if (asarPath) { // already calculated
-      return process.nextTick(() => callback(asarPath));
-    }
-
-    let apmFolder = path.resolve(__dirname, '..');
-    let appFolder = path.dirname(apmFolder);
-    if ((path.basename(apmFolder) === 'ppm') && (path.basename(appFolder) === 'app')) {
-      asarPath = `${appFolder}.asar`;
-      if (fs.existsSync(asarPath)) {
-        return process.nextTick(() => callback(asarPath));
+  getResourcePath() {
+    return new Promise((resolve, _reject) => {
+      if (process.env.ATOM_RESOURCE_PATH) {
+        return void process.nextTick(() => resolve(process.env.ATOM_RESOURCE_PATH));
```

Comment:
There are two answers here, depending on what you compare to.

If you compare to `justMakeTheCall(); return;`, it's really the same thing, kind of stylistics. I tend to like the aesthetics and terseness of it, especially for arrow functions where they are "a clear win" regarding syntax. Some defined style guide could trump it at any second.

If you compare to `return justOneMoreCall();`, it is indeed "just" to ensure that we return `undefined` - but I don't find that so much of a "just" thing. A more clever static analyzer, or Typescript in particular, could give the users of a certain function type hints about the returned value. To let them know on all possible layers that the function is never ever going to produce any useful value, is a noble and sensible goal in my opinion. By applying this convention, it's immediately clear what statement is there to actually produce a value. This is kind of the opposite of the old Coffeescript way that deliberately didn't distinguish between returning values and just ending a function.

Reply:
I personally prefer `justMakeTheCall(); return;`.

`return void justMakeTheCall();` causes some mental gymnastics for me, since `void` is so very uncommon in JavaScript.

It also feels overloaded to me: the `return` and `void justMakeTheCall();` are kind of unrelated, other than it shows the author's intend to to `justMakeTheCall()` as the last thing in the function body before exiting. I rather think `justMakeTheCall(); return;` is a much more conventional (and therefore readable) way to signal this in JavaScript.

Few people work on ppm, and I'm now up to speed on what `void` means in JS, so as one of few people working on this code I am not so opposed to it that I would insist on removing it. But part of me almost feels I should, in case anyone new ever hopes to contribute to ppm. I think conventional and usual-looking code would be more accessible to newcomers to the codebase.

**In summary: I wouldn't insist on changing to `doSomething(); return`, but I would suggest it be done nonetheless.** If it's not done in this PR and if I still feel strongly enough about it later, I can do a follow-up PR to tweak the code not to use `void`.

Reply:
I find the "commonness argument" fallacious because one kind of has to assume a practical reason behind something being uncommon, or else it's really just catch-22. I don't have a representative study down in my pocket but I don't think it's _that_ obscure anyway, it seems like the kind of thing you can read once and remember. Unfortunately, one can come across `return iOnlyReallyExecutedThisOneForTheSideEffects()` which is basically the semantically wrong version of it, and as I said, this is mostly an opt-out from that pattern.

Having said that, there could be extensions to a written or conventional style guide for code here. This isn't my project so I tried to preserve apparent conventions, even the ones I find just silly. For example, I quite vehemently oppose the famous `===` dogma (as it doesn't actually address typing problems, only confuses things further) or, what was even weirder in this code: using parens around basically all comparisons when used in combination with logic operators, as if it was Pascal. It's really not my business to either set guidelines or to challenge existing ones, I only did what felt like a clear win within the established conventions. Since the former status quo was to just return ad-hoc values even for practically void functions, I think eliminating that was a clear win even at the cost of having to learn one pretty old operator. :P

At the end of the day, though, while I wouldn't deliberately sabotage existing conventions, I don't want to put any effort into migrating my code to a previously non-existent convention I softly disagree with. It's probably pretty easy; I can imagine that there is a good enough regex to do 90% of that work but I think that's all the more reason to see who says "it's worth my time". I wouldn't say that, for one.

Reply:
You do have some points about weird conventions that already exist here, even if they aren't our doing. But I suppose you have a point that it isn't the hardest to learn as new, 

Although at the end of the day, I suppose we leave this one be, since you obviously feel quite strong, and like it's been mentioned this isn't a blocker, so I'm willing to say we bank off that to see how soon we can get this in